### DISCUSSION TITLE
v0.2.1+11

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#297] [Announcements]
## Alpha Release [v0.2.1+11]

The day has come for another new update! This update adds a lot of things, and fixes a lot of issues that it is hard to summarize in one page. Before I go on to the update notes, I would like to give a big big big shoutout to all the contributors that have been helping for this past week.

This update was an accumulation of all their work, along with some of my usual work. These contributors include but are not limited to @bactaholic, @micahmo, @Fmstrat, @CTalvio, @ajsosa, @guigs4, @coslu, @minicit! In addition to that, there have been others who have been helping with clarifying issues, bringing up ideas, and providing feedback. Some of the names to shout out are @machinaeZER0, @krestenlaust, @Poolitzer

Please let me know if I have missed anyone :D Now onto the update itself.

**Important: it is highly recommended to reinstall the app for the best (and least buggy) experience possible**

This update brings a few major features:
- Swipe customization - you can now assign different actions to the left and right swipes
- Font scaling - you can now set a font size for titles and content
- Various UI improvements - you'll notice various tweaks which should improve the overall look and experience of Thunder
- Addition of comment sorting - you can now sort comments from any of the available options
- More sharing options - you can now share posts, links, and media from the dialog menu when you long-press on a post
- Improved performance to long comment threads - you should experience less lag as you scroll through long comment threads

As always, this is an alpha release, so do expect bugs to pop up here and there. Without further ado, here are the full changes:

### Added
- Added colour to username in comments to distinguish your own comments
- Added option to allow text preview on text posts in normal view - contribution from @coslu
- Added option to share media link or external URL links in the dialog popup on post long press
- Added back featured post icon on posts within a community
- Added ability to customize swipe gestures on comments and posts
- Added ability to develop with hot reload using docker - contribution from @Fmstrat
- Added initial implementation for font size scaling for titles and content
- Added comment sorting - contribution from @guigs4
- Added option to show title before content - contribution from @Fmstrat
- Added option to show images edge-to-edge - contribution from @Fmstrat
- Addition of manual refresh icons to different screens - contribution from @bactaholic
- Addition of user and community banners - contribution from @CTalvio
- Added ability to share media directly - contribution from @micahmo
- Added initial ability to block community
- Comments now show the number of direct replies - contribution from @micahmo
- Added minor UI changes to the post view - contribution from @CTalvio

### Changed
- Adjusted visual feedback in comment swipe gestures
- Added option in settings to enable/disable swipe on bottom navigation bar to open sidebar on feed page - contribution from @bactaholic
- Added option in settings to enable/disable double-tap on bottom navigation bar to open sidebar on feed page - contribution from @bactaholic
- Decreased scroll distance needed in order to fetch more comments from a thread
- Improved error messages when more comments fail to fetch within a post
- Decreased number of comments to fetch at a time to improve loading performance
- Posts now load first before comment threads to make viewing posts more responsive
- Slight changes to contrast of text in posts - contribution from @Fmstrat
- Added tooltip labels on sort types - contribution from @micahmo
- UI improvements to vote indication in feed view - contribution from @CTalvio
- Community pages now use their display name - contribution from @CTalvio

### Fixed
- Potentially fixed issues with HTTPS certificate errors when running on Android using Adguard with HTTPS filtering enabled
- Improved performance for comment threads with a lot of comments
- Fixed issue where the last comment on a thread could potentially not show up
- Fixed issue where markdown links were not respecting "open in external browser" option - contribution from @Fmstrat
- Fixed issue with download media not working on Android for some users - contribution from @minicit
- Fixed issue where media download button appearance in light mode - contribution from @coslu
- Fixed issue where commenting would bring you out of the post
- Fixed issue where you could not edit your comment in some instances when using your display name
- Fixed issue with duplicate posts being shown on the feed - contribution from @ajsosa
- Increased NSFW blur - contribution from @guigs4
- Fixed issue where longer comment threads would not show up properly
- Fixed a bug where the URI was not parsed to lowercase before checking extensions to parse image dimensions - contribution from @Fmstrat
- Fixed issue where not all comments would show up on the profile page - contribution from @ajsosa

and as usual, even more changes under the hood which will make way for future features.

----
If you find any issues or would like to suggest features, please open up a new issue! As always, contributions are always welcomed here.

<hr /><em>This discussion was created from the release <a href='https://github.com/hjiangsu/thunder/releases/tag/v0.2.1%2B11'>v0.2.1+11</a>.</em>

**Comment by kovalensky** [comment]
Keep in mind that this release logs out from your accounts

**Comment by krestenlaust** [comment]
Thanks for the shout-out. It's been a blast contributing to my favorite app!
